src/charm_state.py

```diff
@@ -196,9 +183,8 @@ def from_charm(
         Raises:
             CharmConfigInvalidError: if the charm configuration is invalid.
         """
-        synapse_config = {k: v for k, v in charm.config.items() if k in KNOWN_CHARM_CONFIG}
         try:
-            valid_synapse_config = SynapseConfig(**synapse_config)  # type: ignore
+            valid_synapse_config = SynapseConfig(**dict(charm.config.items()))  # type: ignore
```

Comment:
Is the `type: ignore` still needed? If so, what is the reason for needing it please?

Reply:
Sure, this is the message:
```python
lint run-test: commands[5] | mypy /home/amanda/Canonical/synapse-operator/src/ /home/amanda/Canonical/synapse-operator/tests/
src/charm_state.py:191: error: Argument 1 to "SynapseConfig" has incompatible type "**dict[str, str]"; expected "bool"  [arg-type]
src/charm_state.py:191: error: Argument 1 to "SynapseConfig" has incompatible type "**dict[str, str]"; expected "int | None"  [arg-type]
Found 2 errors in 1 file (checked 33 source files)
```